scripts/make-dist.sh

```diff
@@ -4,23 +4,18 @@
 # in the LICENSE file.
 set -e
 
-package_name=singularity
-package_version=`(git describe --match 'v[0-9]*' --always 2>/dev/null || cat VERSION 2>/dev/null || echo "") | sed -e "s/^v//;s/-/_/g;s/_/-/;s/_/./g"`
-package_version_short=`(git describe --abbrev=0 --match 'v[0-9]*' --always 2>/dev/null || cat VERSION 2>/dev/null || echo "") | sed -e "s/^v//;s/-/_/g;s/_/-/;s/_/./g"`
+# spec file needs to be at the root of the project
+cp dist/rpm/singularity.spec .
 
-# Remove Dist tarball if it exists
-if [ -f $package_name-$package_version_short.tar.gz ]; then
-    rm -f $package_name-$package_version_short.tar.gz
-fi
+package_name=singularity
+package_version_short="`sed -n 's/^Version: //p' singularity.spec`"
+tree_version="$package_version_short-`sed -n 's/^Release: \([^%]*\).*/\1/p' singularity.spec`"
 
 echo " DIST setup VERSION: $tree_version"
 echo $tree_version > VERSION
-git add VERSION
-# spec file needs to be at the root of the project
-cp dist/rpm/singularity.spec .
-git add singularity.spec
-echo " DIST create tarball: $package_name-$package_version_short.tar.gz"
-git archive --format=tar --prefix=$package_name/ `git stash create` -o $package_name-$package_version_short.tar
-git reset VERSION singularity.spec
-gzip $package_name-$package_version_short.tar
-
+tarball="$package_name-$package_version_short.tar.gz"
+echo " DIST create tarball: $tarball"
+rm -f $tarball
+(echo VERSION; echo singularity.spec; git ls-files) | \
+        tar -T - --xform "s,^,$package_name/," -czf $tarball
```

Comment:
--xform is not supported on non GNU tar. That's why I preferred letting git generate the archive.

Can you achieve the same result with standard tools?